extensions/interactions/ItemSelectionInput/protractor.js

```diff
@@ -0,0 +1,159 @@
+  ).setValue(maxSelectionAllowed);
+
+  forms.ListEditor(elem).setLength(richTextInstructionsArray.length);
+  for (var i = 0; i < richTextInstructionsArray.length; i++) {
+    var richTextEditor = forms.ListEditor(elem).editItem(i, 'RichText');
+    richTextEditor.clear();
+    richTextInstructionsArray[i](richTextEditor);
+  }
+};
+
+// These members of richTextInstructionsArray each describe how to check one of
+// the options.
+var expectInteractionDetailsToMatch = function(
+    elem, richTextInstructionsArray) {
+  elem.all(by.repeater('choice in $ctrl.choices track by $index'))
+    .then(function(optionElements) {
+      expect(optionElements.length).toEqual(richTextInstructionsArray.length);
+      for (var i = 0; i < optionElements.length; i++) {
+        forms.expectRichText(optionElements[i].element(by.css(
+          '.protractor-test-item-selection-option'
+        ))).toMatch(richTextInstructionsArray[i]);
+      }
+    });
+};
+
+// type of object returned by interaction
+var answerObjectType = 'SetOfHtmlString';
+
+// 'elem' is the HTML element containing the form to submit the answer to.
+// 'answer' Set([{String}]) is the text on the multiple-choice item to select.
+var submitAnswer = function(elem, answer) {
+  var answerArray = Array.from(answer);
+
+  for (var i = 0; i < answerArray.length; i++) {
+    var desiredAnswer = answerArray[i];
+    elem.element(by.cssContainingText(
+      '.protractor-test-item-selection-input-item', desiredAnswer))
+      .element(by.css(
+        '.protractor-test-item-selection-input-checkbox')).click();
+  }
+
+  var submitAnswerButton = element(by.css(
+    '.protractor-test-submit-answer-button'));
+  waitFor.elementToBeClickable(submitAnswerButton,
+    'Submit Answer button is not clickable');
+  submitAnswerButton.click();
+};
+
+var testSuite = [{
+  interactionArguments: [[function(editor) {
+    editor.appendBoldText('answer1');
+  }, function(editor) {
+    editor.appendItalicText('answer2');
+  }, function(editor) {
+    editor.appendItalicText('answer3');
+  }], 3],
+  ruleArguments: ['DoesNotContainAtLeastOneOf', ['answer1', 'answer2']],
+  expectedInteractionDetails: [[function(checker) {
+    checker.readBoldText('answer1');
+  }, function(checker) {
+    checker.readItalicText('answer2');
+  }, function(checker) {
+    checker.readItalicText('answer3');
+  }]],
+  wrongAnswers: [['answer1', 'answer2']],
+  correctAnswers: [['answer3']]
+}, {
+  interactionArguments: [[function(editor) {
+    editor.appendBoldText('answer1');
+  }, function(editor) {
+    editor.appendItalicText('answer2');
+  }, function(editor) {
+    editor.appendItalicText('answer3');
+  }], 3],
+  ruleArguments: ['Equals', ['answer1', 'answer2']],
+  expectedInteractionDetails: [[function(checker) {
+    checker.readBoldText('answer1');
+  }, function(checker) {
+    checker.readItalicText('answer2');
+  }, function(checker) {
+    checker.readItalicText('answer3');
+  }]],
```

Comment:
This code gets repeated a lot. Can we reduce the redundancy?

Reply:
Done!